### ISSUE TITLE
Compatibility with Older CUDA Versions

### ISSUE THREAD

**Issue by brycelelbach** [NVIDIA/thrust#916]
After reading #915 (go do that now!), you may be wondering about Thrust's compatibility with older versions of CUDA. Historically, most of Thrust's CUDA backend has worked with older versions of CUDA, with some features requiring a particular version of CUDA to work (ditto for C++ standards). However, as far as I know, there has never been a firm policy about how far back CUDA compatibility goes.

With this new scheme, will Thrust v15.0.0 only work with CUDA 15.0? No. Going forward, Thrust will support [0] the two CUDA major release series [1] prior to the series that version of Thrust was released with.

So, Thrust v15.0.0 would be released with CUDA 15.0 and on GitHub, and would support CUDA 13.y and CUDA 14.y.

Some major features of Thrust may require newer  versions of CUDA

[0] Thrust supporting an older version of CUDA means:
* You should be able to use GitHub Thrust with that version of CUDA, although some features may be unavailable.
* When we know something will not work with a supported version of CUDA, we either (a) provide an `#ifdef`'d legacy implementation or (b) make the feature require a newer CUDA version and provide a suitable compile-time diagnostic (done when legacy support is hard). When using Thrust with a supported version of CUDA, either your code works or you get a compilation diagnostic explaining which feature will not work with your CUDA version.
* If the last bullet point is violated and you file an issue, we will treat it as bug.

Thrust supporting a version of CUDA does not mean:
* That our continuous integration system will build or test against that version of CUDA.
* That we will prioritize or commit to fixing all bugs relating to support.
* That all major features will work with that version of CUDA; it does mean those features should tell you at compile time if they require a newer CUDA version.

[1]: A CUDA major release series is a set of CUDA releases with the same major version number (CUDA 7.* series == 7.0 7.5, CUDA 8.* series == 8.0, CUDA 9.* series == 9.0 9.1 9.2).

TL;DR
=====

**Assuming we proceed with #915:**

* **Going forward, Thrust vX.Y.Z is the version released with CUDA X.Y (#915).**
* **Thrust vX.Y.Z also supports CUDA (X-2).Y and CUDA (X-1).Y; this means features should either work or tell you at compile time that they won't work with these CUDA versions.**

**Comment by brycelelbach** [comment]
@jaredhoberock please review.

**Comment by jrhemstad** [comment]
Closing in favor of https://github.com/NVIDIA/cccl/issues/66
